rocketpy/simulation/monte_carlo.py

```diff
@@ -359,6 +371,22 @@ def __export_flight_data(
             for export_item in self.export_list
         }
 
+        if self.export_function is not None:
+            try:
+                additional_exports = self.export_function(flight)
+            except Exception as e:
+                raise ValueError(
+                    "An error was encountered running your custom export function. "
+                    "Check for errors in 'export_function' definition."
+                ) from e
+
+            for key in additional_exports.keys():
+                if key in self.export_list:
+                    raise ValueError(
+                        f"Invalid export function, returns dict which overwrites key, '{key}'"
+                    )
```

Comment:
We only need to do this once, I recommend moving this validation to somewhere in the init method so we don't need to re-run it every time.

Reply:
> It is a good idea to run this only once, but moving it to the init method might be problematic in the current implementation. The reasoning is as follows: to validate that there are no keys being overwritten, we have to call `export_list` so we can have the dictionary outputted by the callback. However, in order to make that call, we have to have the data from the simulations. I see two options:
> 
> 1. Create a "mock dictionary", or at least its keys, from `export_function` and make the test in the init method;
> 2. Create a flag to check if `export_function` was validated. On the first call of these lines (inside `__export_flight_data`, we make the test and set the flag, hence halting further validations.

@Lucas-Prates if we have a `__check_export_list` method, I think we should also have a `__check_export_function` as well. I like the idea of a `mock_flight`, but I don't know how hard it would be to make it happen.

I believe my other comment is related to this. If instead of a "callback function" that evaluates and returns a dictionary, maybe we should receive a dictionary with different callback functions. That way we could easily get the expected key names.

Other option is to not make any validation, let the user deal with the fact they will be overwriting the values if use the same name of variables in the dictionary.

Reply:
I like the idea of separating the callback functions. Thinking more carefully, it would even allow for more descriptive error raises when there is an implementation error by the user.

Reply:
My two cents on the matter:

1. My preference would be not to validate and document that `export_function` overwrites identical keys of `export_list`;
2. Another option would be using the `flight` parameter from `__init__` as the discussed `mocked_flight`, it would also compute whether the `export_function` as a whole works.